flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleServiceLoader.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.shuffle;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.util.FlinkException;
+import org.apache.flink.util.InstantiationUtil;
+
+import java.util.Optional;
+
+import static org.apache.flink.configuration.CoreOptions.SHUFFLE_SERVICE;
+
+/**
+ * Utility to load the pluggable {@link ShuffleService} implementations.
+ */
+public class ShuffleServiceLoader {
+
+	private ShuffleServiceLoader() {
+	}
+
+	public static ShuffleService<?, ?, ?> loadShuffleService(Configuration configuration) throws FlinkException {
+		String shuffleServiceClassName = configuration.getString(SHUFFLE_SERVICE);
+		Optional<ShuffleService<?, ?, ?>> shuffleService = lookupEmbeddedShuffleServices(shuffleServiceClassName);
+		return shuffleService.orElseThrow(() -> new FlinkException(
+			String.format(
+				"Could not instantiate the ShuffleService '%s'. " +
+					"Please make sure that this class is on your class path.",
+				shuffleServiceClassName)));
+	}
+
+	private static Optional<ShuffleService<?, ?, ?>> lookupEmbeddedShuffleServices(String shuffleServiceClassName) {
+		ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+		try {
+			return Optional.of((ShuffleService<?, ?, ?>) InstantiationUtil.instantiate(
+				shuffleServiceClassName,
+				ShuffleService.class,
+				classLoader));
+		} catch (ClassNotFoundException e) {
+			return Optional.empty();
```

Comment:
Why are we swallowing this exception here? It looks as if we recreate it at the call site of `lookupEmbeddedShuffleServices` if the result is `empty`. I think it would be better to simply return `ShuffleService<?, ?, ?>` and to allow this method to throw a `ClassNotFoundException`.